hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OzoneFSUtils.java

```diff
@@ -205,4 +206,25 @@ public static int getFileCount(String keyName) {
     java.nio.file.Path keyPath = Paths.get(keyName);
     return keyPath.getNameCount();
   }
+
+  /**
+   * Returns true if the bucket is FS Optimised.
+   * @param bucket
+   * @return
+   */
+  public static boolean isFSOptimizedBucket(OmBucketInfo bucket) {
+    // layout version V1 represents optimized FS path
```

Comment:
@sadanand48, nice idea to add this to utility function. Can you please do a few changes to avoid duplicate functionality exists in [BasicOzoneClientAdapterImpl#L552](https://github.com/apache/ozone/blob/HDDS-2939/hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java#L552).

Can you do changes like:

 ```
OzoneFSUtils.java

 /**
   * Returns true if the bucket is FS Optimised.
   * @param buckeMetadata
   * @return
   */
  public static boolean isFSOptimizedBucket(Map<String, String> buckeMetadata) {
    // layout version V1 represents optimized FS path
    boolean layoutVersionEnabled =
            org.apache.commons.lang3.StringUtils.equalsIgnoreCase(
                    OMConfigKeys.OZONE_OM_LAYOUT_VERSION_V1,
                    buckeMetadata.get(OMConfigKeys.OZONE_OM_LAYOUT_VERSION));

    boolean fsEnabled =
            Boolean.parseBoolean(buckeMetadata
                    .get(OMConfigKeys.OZONE_OM_ENABLE_FILESYSTEM_PATHS));

    return layoutVersionEnabled && fsEnabled;
  }
```

```
BasicOzoneClientAdapterImpl.java

  @Override
  public boolean isFSOptimizedBucket() {
    // layout version V1 represents optimized FS path
    return OzoneFSUtils.isFSOptimizedBucket(bucket.getMetadata());
  }

```

Reply:
Done.